tests/test_preconditioning.py

```diff
@@ -0,0 +1,114 @@
+# pylint: disable = import-error, missing-class-docstring, missing-function-docstring, missing-module-docstring
+import unittest
+from json_journal import preconditioning
+
+
+class TestSortLedger(unittest.TestCase):
+
+    def test_happypath(self):
+        # Given
+        ledger = [
+            {
+                "date": "2022-11-01",
+                "description": "bills",
+                "entries": [
+                    {"account": "expense:bills", "amount": 650},
+                    {"account": "assets:bank", "amount": -650}
+                ]
+            },
+
+            {
+                "date": "2022-12-03",
+                "description": "gym membership",
+                "entries": [
+                    {"account": "expense:gym", "amount": 10},
+                    {"account": "assets:bank", "amount": -10}
+                ]
+            },
+
+            {
+                "date": "2022-11-10",
+                "description": "savings",
+                "entries": [
+                    {"account": "assets:savings", "amount": 50},
+                    {"account": "assets:bank", "amount": -50}
+                ]
+            }
+        ]
+
+        expected_ledger = [
+            {
+                "date": "2022-11-01",
+                "description": "bills",
+                "entries": [
+                    {"account": "expense:bills", "amount": 650},
+                    {"account": "assets:bank", "amount": -650}
+                ]
+            },
+
+            {
+                "date": "2022-11-10",
+                "description": "savings",
+                "entries": [
+                    {"account": "assets:savings", "amount": 50},
+                    {"account": "assets:bank", "amount": -50}
+                ]
+            },
+
+            {
+                "date": "2022-12-03",
+                "description": "gym membership",
+                "entries": [
+                    {"account": "expense:gym", "amount": 10},
+                    {"account": "assets:bank", "amount": -10}
+                ]
+            }
+        ]
+
+        # When
+        actual_ledger = preconditioning.sort_journal_by_date(ledger)
+
+        # Then
+        self.assertEqual(expected_ledger, actual_ledger)
+
+
+class TestEntriesSumToZero(unittest.TestCase):
```

Comment:
Is it worth adding a test for the error case?

Reply:
done